lib/ocpp/v201/charge_point.cpp

```diff
@@ -218,8 +229,13 @@ void ChargePoint::on_transaction_finished(const int32_t evse_id, const DateTime&
         id_token_opt = _id_token;
     }
 
+    std::optional<bool> offline;
+    if (this->websocket_connection_status == WebsocketConnectionStatusEnum::Disconnected) {
```

Comment:
nit: I would put this into a separate function because this is the same code 6 times. We could also change the signature of transaction_event_req to not take a std::optional<bool> but just a bool instead and always report true or false in every TransactionEvent.req message

Reply:
My 2 cents on this: Only send the `offline` value if we were offline and otherwise omit it. All the functionality is there and it saves a bit of data, even if its a minor amount.

You could indeed change it to a bool in transaction_event_req and do the omitting in that function.